### PR TITLE
Add the termination checker to the pipeline

### PR DESCRIPTION
**Pull request by jonaprieto**
After merging this PR, any compilation of a MiniJuvix program should pass the termination checker. This check takes during the Abstract-to-MicroJuvix translation.  For example, the compiler would complain about nontermination with the following program, unless we use the global flag `--no termination`  or the keyword `terminating`.

```bash
$ cd tests/negative/Termination
$ cat ToEmpty.mjuvix 
module ToEmpty;
  axiom A : Type;
  inductive Empty {};
  f : A -> Empty;
  f x := f x;
end;
```

Then,

```
$ minijuvix minic ToEmpty.mjuvix
f Fails the termination checking
```

As said above, to disable this check we can run for example:

```bash
$ minijuvix minic ToEmpty.mjuvix --no-termination
```

### CODE REVIEW

**Comment by jonaprieto** [comment]
Blocked by #117

File: src/MiniJuvix/Pipeline/EntryPoint.hs [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-  { _entryRoot :: FilePath,
-    _entryModulePaths :: NonEmpty FilePath
+  { _entryPointRoot :: FilePath,
+    _entryPointNoTermination :: Bool,
+    _entryPointModulePaths :: NonEmpty FilePath
   }
   deriving stock (Eq, Show)
 
+defaultEntryPoint :: FilePath -> EntryPoint
+defaultEntryPoint mainFile = EntryPoint "." False (pure mainFile)
```
Comment (janmasrovira) [inline src/MiniJuvix/Pipeline/EntryPoint.hs:17]: now that we are adding more fields, I think it makes sense to use record syntax with explicit fields

**Review by janmasrovira — CHANGES_REQUESTED** [review]
Some small comments

**Review by janmasrovira — APPROVED** [review]
🚢

**Pull request merged**